File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head==None:
            return False
        l=head
        r=head
        while(r is not None and r.next is not None):
            l=l.next
            r=r.next.next
            if(l==r):
                return True
        return False

File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        if(head==null){
            return false;
        }
        ListNode l=head;
        ListNode r=head;
        while(r!=null && r.next!=null){
            l=l.next;
            r=r.next.next;
            if(l==r){
                return true;
            }
        }
        return false;
    }
}